### PR TITLE
set column widths and date format

### PR DESCRIPTION
**Pull request by mark-prins**
This sets columns widths based on header and row content, using the current font in the template it looks great. There's no guarantee this will always be the case, but it works for now.

Have also formatted dates - which are helpfully returned as float64 values. The parsing is a bit iffy, but once again, it works.

You'll need to update the template ( attached ) as I've merged the date cell so that it looks fine when column 1 is not so wide.

[template.xlsx](https://github.com/openmsupply/msupply-dashboard-app/files/5620404/template.xlsx)

### CODE REVIEW

**Comment by mark-prins** [comment]
and here's how it looks for me... 
[Expiring Items.xlsx](https://github.com/openmsupply/msupply-dashboard-app/files/5620410/Expiring.Items.xlsx)

File: backend/pkg/reporter/reporter.go [hunk 1]
```diff
@@ -218,6 +244,48 @@ func (r *Report) writeRows(sheetName string, rows [][]interface{}) error {
 	return nil
 }
 
+func (r *Report) setColumnWidths(sheetName string, columns []api.Column, rows [][]interface{}) error {
+
+	headerFontCorrectionFactor := 1.35
+	maximumContentLengths := make(map[int]float64)
+
+	if len(columns) > 0 {
```
Comment (joshxg) [inline backend/pkg/reporter/reporter.go:252]: if seems redundant?

**Review by joshxg — APPROVED** [review]
Wicked!

image

Couple minor comments

- Slightly off when someone is annoying and uses a string with all caps, it seems. Not a big one
- Maybe commit the template?

File: backend/pkg/reporter/reporter.go (hunk 1 above)
Comment (mark-prins) [reply backend/pkg/reporter/reporter.go:252]: thanks, removed

**Comment by mark-prins** [comment]
awesome! thanks @joshxg - both for the review and the inspiration to test against the `[transact]` table. 
found that the dates are coming through as strings for that table, though `[item_line]` is giving me float64 even though both are `date` types. End result is that we have a little more date handling 😄 

have vertically aligned the numbers, fixed the heading padding and catered for ALL UPPERCASE STRING VALUES TOO.

**Comment by joshxg** [comment]
PERFECT

**Pull request merged**